### PR TITLE
Add initial detection of WMDS, and improve detection of MediaMax CD-3

### PR DESCRIPTION
**Pull request by SilasLaspada**
Add initial detection of Windows Media Data Session, and improve detection of MediaMax CD-3.

### CODE REVIEW

File: BurnOutSharp/ProtectionType/WMDS.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                return null;
+
+            // Found on "All That I Am" by Santana (Barcode 8 2876-59773-2 6)
+            string name = pex.FileDescription;
+            if (!string.IsNullOrWhiteSpace(name) && name.StartsWith("Windows Media Data Session Licensing Engine", StringComparison.OrdinalIgnoreCase))
+                return $"Windows Media Data Session DRM";
+
+            return null;
+        }
+        /// <inheritdoc/>
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/WMDS.cs:31]: Nit: Space before this line

File: BurnOutSharp/ProtectionType/WMDS.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+namespace BurnOutSharp.ProtectionType
+{
+    // The Windows Media Data Session Toolkit was created as a form of DRM for audio CDs (and supposedly DVDs, but with no standard for multiple sessions existing for pressed DVDs, this seems unlikely), which ultimately acts a framework
+    // It seems to provide a framework for standardized storing of protected music, as well as allowing the disc publisher to specify how many copies they wish users to be able to create using Windows Media Player.
+    // Known to be used along with MediaMax CD-3, possibly others.
+    // Reference: https://news.microsoft.com/2003/01/20/microsoft-releases-new-windows-media-data-session-toolkit-enabling-second-session-creation/
+    public class WMDS : IPathCheck, IPortableExecutableCheck
+    {
+
+        public string CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/WMDS.cs:17]: Nit: No space before this line

File: BurnOutSharp/ProtectionType/MediaMaxCD3.cs [hunk 3]
```diff
@@ -65,6 +68,15 @@ public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<strin
             var matchers = new List<PathMatchSet>
             {
                 new PathMatchSet(new PathMatch("LaunchCd.exe", useEndsWith: true), "MediaMax CD-3"),
+
+                // Found on "All That I Am" by Santana (Barcode 8 2876-59773-2 6)
+                new PathMatchSet(new List<PathMatch>
+                {
+                    // TODO: Verify if these are OR or AND
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/MediaMaxCD3.cs:75]: If you have them in the same set, I think, by default, it "AND"s them

File: BurnOutSharp/ProtectionType/WMDS.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        /// <inheritdoc/>
+        public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string> files)
+        {
+            var matchers = new List<PathMatchSet>
+            {
+                // Found on "All That I Am" by Santana (Barcode 8 2876-59773-2 6)
+                new PathMatchSet(new List<PathMatch>
+                {
+                    // TODO: Verify if these are OR or AND
+                    new PathMatch("LicGen.exe", useEndsWith: true),
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/WMDS.cs:40]: Is `LicGen.exe` too vague for the file path check?

Comment (SilasLaspada) [reply BurnOutSharp/ProtectionType/WMDS.cs:40]: Personally, I was definitely concerned about it being too vague for that.

File: BurnOutSharp/ProtectionType/MediaMaxCD3.cs (hunk 3 above)
Comment (mnadareski) [reply BurnOutSharp/ProtectionType/MediaMaxCD3.cs:75]: Eh, fair enough. I've had long-lived TODOs in the past.

File: BurnOutSharp/ProtectionType/WMDS.cs (hunk 4 above)
Comment (mnadareski) [reply BurnOutSharp/ProtectionType/WMDS.cs:40]: Have you seen it anywhere else yet?

**Review by mnadareski — APPROVED** [review]

**Pull request merged**